Hide add permission button in non beta and for system roles.

Hyperkid123 opened the pull request:
part of: https://issues.redhat.com/browse/RHCLOUD-10977
fixes: https://issues.redhat.com/browse/RHCLOUD-11126

### Changes
- hide the add permission in non beta env
- hide the add permission button for system roles

It is missing from acceptance criteria but belongs to custom roles stuff (thanks @john-dupuy). Waiting for confirmation that we should also hide this behind the beta flag.

john-dupuy commented on src/smart-components/role/role-permissions.js:
```diff
@@ -141,45 +141,47 @@ const Permissions = () => {
+      },
+    },
+    {
+      label: 'Remove',
```
Is this part also gated by `beta`? If not it should be.

Hyperkid123 replied:
No only the first item in the array.

Hyperkid123 replied:
It's still visible on small screens! Nice catch.

john-dupuy replied:
Oh interesting, for some reason the dropdown is hidden on large screens.

Hyperkid123 replied:
It's a bit weird. I think I'll have to look into that component. Because if you now have both the add permission and remove, you can't remove the selected permission on a large screen because the dropdown is just not there...

Hyperkid123 replied:
So it won't affect prod but in prod beta users won't be able to remove permissions. Here's a question, should they be even able to remove permissions in prod? They can't add any so I would find it strange if users could remove them.

john-dupuy replied:
Yeah, this is a part of https://issues.redhat.com/browse/RHCLOUD-11126 btw

john-dupuy replied:
I think the gating for the "Remove" dropdown item should be the same as for the "Add permissions" button:
* only present for non-system roles in beta environments

Hyperkid123 replied:
Right. It looks like missing CSS. The element is in the DOM. Only it has a `display: none` CSS property. After fixing it in the browser console it shows properly. I think I will try and fix it on Monday. And before we merge to prod.

> I think the gating for the "Remove" dropdown item should be the same as for the "Add permissions" button

Agreed.

Hyperkid123 commented:
@john-dupuy alright. The buttons are now properly hidden and the dropdown is now visible.

john-dupuy commented:
@Hyperkid123 thanks for this, checking this PR locally I noticed that in the `beta` environment on the roles page that the dropdown has no items in it. Screenshot below.
![Screenshot from 2021-01-04 08-41-47](https://user-images.githubusercontent.com/44065123/103541381-589aba80-4e69-11eb-8082-78704679560c.png)

Additionally, I noticed that in non-beta environments - users are still able to edit and delete custom roles via the dropdown present on a row within the table. Similarly on the role details page, a user is able to individually remove permissions using this dropdown. Do we want to also gate these dropdowns?

Hyperkid123 commented:
> the dropdown has no items in it
This will be a bug in FEC. I bet when you shrink the window the dropdown will be populated with that one item (Create role). We need to fix it there.

>  users are still able to edit and delete custom roles via the dropdown present on a row within the table
We gate those as well. (I hate feature flags)

Hyperkid123 commented:
@john-dupuy 
> users are still able to edit and delete custom roles via the dropdown present on a row within the table

On second thought. If users can create custom roles, there should be no custom roles in the table so I don't really see why to gate these options. (Yes I know the API is avaiable)

john-dupuy commented:
> I bet when you shrink the window the dropdown will be populated with that one item (Create role).

Yep, just confirmed this was the case. I guess eventually we could put a "Remove" button there too? Provided that we also have a way to select multiple custom roles.

Hyperkid123 commented:
I would assume so. I will try and fix it inside FEC, but I would not block this by that. This issue present in each app, but nobody noticed it probably.

john-dupuy commented:
> If users can create custom roles, there should be no custom roles in the table so I don't really see why to gate these options

Yeah that's a good point. Custom roles could be in the table if a user makes them in a beta environment however and goes back to the non-beta environment. We might hit an edge case where someone removes the custom role accidentally in the non-beta environment and is confused as to why they can't create a new one.

Hyperkid123 commented:
@john-dupuy the empty dropdown is now fixed. I have removed some CSS rules which were hiding the elements. So the dropdown was not actually empty, but the elements were hidden. So it's fortunately not FEC problem.

It's now using the dedicated action instead which was created for this exact use-case.

![dropdown](https://user-images.githubusercontent.com/22619452/103545384-c7940580-4ea1-11eb-8f3c-02d256b42586.gif)
